sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Top.java

```diff
@@ -168,7 +168,40 @@ private Top() {
    * {@code PCollection} of {@code KV}s and return the top values associated with each key.
    */
   public static <T extends Comparable<T>> Combine.Globally<T, List<T>> largest(int count) {
-    return Combine.globally(new TopCombineFn<>(count, new Natural<T>()));
+    return Combine.globally(largestFn(count));
+  }
+
+  /** Returns a {@link TopCombineFn} that aggregates the largest count values. */
+  public static <T extends Comparable<T>> TopCombineFn<T, Natural<T>> largestFn(int count) {
+    return new TopCombineFn<T, Natural<T>>(count, new Natural<T>()) {};
+  }
+  /** Returns a {@link TopCombineFn} that aggregates the largest count long values. */
+  public static TopCombineFn<Long, Natural<Long>> largestLongsFn(int count) {
+    return new TopCombineFn<Long, Natural<Long>>(count, new Natural<Long>()) {};
```

Comment:
can java infer more of `<>`? E.g. `new TopCombineFn<>(count, new Natural<Long>()) {};`

Reply:
No, this isn't allowed for anonymous inner classes